### DISCUSSION TITLE
Question on dynamic dosing

### DISCUSSION THREAD

**Discussion by kevin199011** [metrumresearchgroup/mrgsolve#1270] [Q&A]
Hi Kyle, I have a question about dynamic dosing. Say we have a neutropenia PKPD model, and would like to do a trial simulation with it. The dose regimen is every 21 days, and the observation is every 7 days. However, we would like to do an adaptive simulation based on trial design, that patients will be dosed if neutrophil count > 1*10^9/L.
For example:
* For a normal patient, the dose is on day 0, 21, 42 ...
* For a patient developed Gr3 neutropenia, the dose should be: day 0, (day 21 not dosed, neutrophil count < 1), day 28 dosed (neutrophil count > 1), (day 49 not dosed, neutrophil count < 1), day 56 dosed (neutrophil count > 1).

I'm wondering if this could be achieved with the evtool plugin? Thank you!

**Comment by kylebaron** [comment]
Hi @kevin199011  - 

Thanks for the question. You can use evtools to help with this simulation, but IMO it'll be easier (easiest) to write some code yourself to do the "bookkeeping". 

It looks like you are dosing every 21 days and observing every 7; if a dose is held, you would only restart at another observation time once you got back above 1. Is that right? If so, it'll be pretty straightforward to implement this bookkeeping. 

Set up a variable to track when the "next" dose is (TIME + 21 days) when neut > 1 and (TIME + 7 days ) when neut < 1. Be sure to reset the next dose time tracker when a new ID is started. 

Let me know if you think this does (doesn't) capture what you want to do.

Kyle
